ocrd_network/ocrd_network/processing_server.py

```diff
@@ -554,63 +587,59 @@ async def get_processor_job(self, processor_name: str, job_id: str) -> PYJobOutp
 
     async def remove_from_request_cache(self, result_message: PYResultMessage):
-        job_id = result_message.job_id
-        state = result_message.state
+        result_job_id = result_message.job_id
+        result_job_state = result_message.state
         path_to_mets = result_message.path_to_mets
         workspace_id = result_message.workspace_id
 
-        self.log.debug(f"Received result for job with id: {job_id} has state: {state}")
-
-        if state == StateEnum.failed:
-            # TODO: Call the callback to the Workflow server if the current processing step has failed
-            pass
-
-        if state != StateEnum.success:
-            # TODO: Handle other potential error cases
-            pass
-
-        job_db = await db_get_processing_job(job_id)
-        if not job_db:
-            self.log.exception(f"Processing job with id: {job_id} not found in DB")
-        job_output_file_grps = job_db.output_file_grps
-        job_page_ids = expand_page_ids(job_db.page_id)
+        self.log.debug(f"Result job_id: {result_job_id}, state: {result_job_state}")
 
         # Read DB workspace entry
-        workspace_db = await db_get_workspace(
-            workspace_id=workspace_id,
-            workspace_mets_path=path_to_mets
-        )
-        if not workspace_db:
+        db_workspace = await db_get_workspace(workspace_id=workspace_id, workspace_mets_path=path_to_mets)
+        if not db_workspace:
             self.log.exception(f"Workspace with id: {workspace_id} or path: {path_to_mets} not found in DB")
 
-        locked_ws_pages = workspace_db.pages_locked
-        # Update locked pages by unlocking the pages in the request
-        self.unlock_pages(
-            locked_ws_pages=locked_ws_pages,
-            output_file_grps=job_output_file_grps,
-            page_ids=job_page_ids
-        )
-
-        # Update the locked pages dictionary in the database
-        await db_update_workspace(
-            workspace_id=workspace_id,
-            workspace_mets_path=path_to_mets,
-            pages_locked=locked_ws_pages
-        )
-
-        # Take the next request from the cache (if any available)
         workspace_key = workspace_id if workspace_id else path_to_mets
-
+        # Take the next request from the cache (if any available)
         if workspace_key not in self.processing_requests_cache:
             self.log.debug(f"No internal queue available for workspace with key: {workspace_key}")
             return
 
+        if result_job_state == StateEnum.failed:
+            if len(self.processing_requests_cache[workspace_key]):
+                self.log.debug(f"Cancelling jobs dependent to job_id: {result_job_id}")
+                await self.cancel_dependent_jobs(
+                    job_id=result_job_id,
+                    internal_queue=self.processing_requests_cache[workspace_key]
+                )
+
+        if result_job_state != StateEnum.success:
+            # TODO: Handle other potential error cases
```

Comment:
A log statement would be a good first step and helpful for debugging.

Reply:
The resulting `job_id` and `job_state` are already printed (at the beginning) when the callback method is invoked.

Reply:
But I could refactor it, yes.